engine/src/main/java/org/pentaho/di/repository/RepositoryConnectionUtils.java

```diff
@@ -0,0 +1,178 @@
+   * @throws KettleSecurityException
+   * @throws ActionExecutionException
+   */
+  public static Repository connectToRepository( final String repositoryName, boolean isSingleDiServerInstance,
+             IAuthorizationPolicy authorizationPolicy, String userName,
+             String fullyQualifiedServerUrl, LoggingBuffer logBuffer ) throws KettleSecurityException, KettleException,
+              ActionExecutionException {
+
+    if ( !hasExecutePermission( authorizationPolicy ) ) {
+      throw new IllegalStateException(
+        BaseMessages.getString( "RepositoryConnectionUtils.ERROR_0001_NO_PERMISSION_TO_EXECUTE" ) ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( "Creating Meta-repository" );
+    }
+
+    RepositoriesMeta repositoriesMeta = new RepositoriesMeta();
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Populating Meta repository" );
+    }
+
+    try {
+      if ( isSingleDiServerInstance ) {
+        if ( logger.isDebugEnabled() ) {
+          logger.debug( "singleDiServerInstance=true, loading default repository" );
+        }
+
+        // Get the embedded repository xml
+        String repositoriesXml = getEmbeddedDefaultRepositoryXml( fullyQualifiedServerUrl );
+
+        ByteArrayInputStream sbis = new ByteArrayInputStream( repositoriesXml.getBytes( Const.XML_ENCODING ) );
+        repositoriesMeta.readDataFromInputStream( sbis );
+      } else {
+        // TODO: add support for specified repositories.xml files...
+        repositoriesMeta.readData(); // Read from the default $HOME/.kettle/repositories.xml file.
+      }
+    } catch ( Exception e ) {
+      throw new ActionExecutionException( BaseMessages.getString(
+        "RepositoryConnectionUtils.ERROR_0018_META_REPOSITORY_NOT_POPULATED" ), e ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Finding repository metadata" );
+    }
+    // Find the specified repository.
+    RepositoryMeta repositoryMeta = null;
+    try {
+      if ( isSingleDiServerInstance ) {
+        repositoryMeta = repositoriesMeta.findRepository( SINGLE_DI_SERVER_INSTANCE );
+      } else {
+        repositoryMeta = repositoriesMeta.findRepository( repositoryName );
+      }
+
+
+    } catch ( Exception e ) {
+      throw new ActionExecutionException( BaseMessages.getString( PKG,
+        "RepositoryConnectionUtils.ERROR_0004_REPOSITORY_NOT_FOUND", repositoryName ), e ); // $NON-NLS-1$
+    }
+
+    if ( repositoryMeta == null ) {
+      if ( logger.isDebugEnabled() && logBuffer != null ) {
+        logger.debug( logBuffer.getBuffer().toString() );
+      }
+      throw new ActionExecutionException( BaseMessages.getString( PKG,
+        "RepositoryConnectionUtils.ERROR_0004_REPOSITORY_NOT_FOUND", repositoryName ) ); //$NON-NLS-1$
+    }
+
+    if ( logger.isDebugEnabled() ) {
+      logger.debug( " Getting repository instance " ); //$NON-NLS-1$
+    }
+    Repository repository = null;
```

Comment:
**Intellij warning:** Initializing this variable to null value is redundant

Reply:
Same comment as above - I initialized it only because IntelliJ was complaining ..

Reply:
same as above, not a blocker, although it does seem to be redundant and may induce in error

Reply:
Attaching screenshot: 
<img width="789" alt="Screenshot 2019-04-15 at 17 53 00" src="https://user-images.githubusercontent.com/1265796/56150776-88c06c80-5fa7-11e9-917a-054fbeebcbf6.png">

Reply:
![localVariablesScreenshot](https://user-images.githubusercontent.com/17853208/56152656-518b9480-5f82-11e9-87ad-66beaa3150a4.png)